packages/opentelemetry-scope-zone-peer-dep/src/ZoneScopeManager.ts

```diff
@@ -0,0 +1,511 @@
+    if (activeZoneName) {
+      const scopes = this._scopes[activeZoneName];
+      if (scopes) {
+        return scopes[scopes.length - 1];
+      }
+    }
+    return undefined;
+  }
+
+  /**
+   * @param target Function to be executed within the scope
+   * @param scope
+   */
+  private _bindFunction<T extends Function>(target: T, scope?: unknown): T {
+    const manager = this;
+    const contextWrapper = function(this: any, ...args: unknown[]) {
+      return manager.with(scope, () => target.apply(this || scope, args));
+    };
+    Object.defineProperty(contextWrapper, 'length', {
+      enumerable: false,
+      configurable: true,
+      writable: false,
+      value: target.length,
+    });
+    return (contextWrapper as unknown) as T;
+  }
+
+  /**
+   * @param obj target object on which the listeners will be patched
+   * @param scope
+   */
+  private _bindListener<T>(obj: T, scope?: unknown): T {
+    const target = (obj as unknown) as TargetWithEvents;
+    if (target.__ot_listeners !== undefined) {
+      return obj;
+    }
+    target.__ot_listeners = {};
+
+    if (typeof target.addEventListener === 'function') {
+      target.addEventListener = this._patchAddEventListener(
+        target,
+        target.addEventListener,
+        scope
+      );
+    }
+
+    if (typeof target.removeEventListener === 'function') {
+      target.removeEventListener = this._patchRemoveEventListener(
+        target,
+        target.removeEventListener,
+        scope
+      );
+    }
+
+    return obj;
+  }
+
+  /**
+   * Removes references to the targets that were bind to certain scope / span
+   * @param scope
+   */
+  private _cleanEventTargetsFromScope(scope: any, target?: TargetWithEvents) {
+    // console.log('_cleanEventTargetsFromScope1', this._scopesWithEventTargets.length, scope && scope.name);
+    for (let i = this._scopesWithEventTargets.length - 1; i >= 0; i--) {
+      const scopeWithEventTargets = this._scopesWithEventTargets[i];
+      if (scopeWithEventTargets.scope === scope) {
+        if (
+          typeof target !== 'undefined' &&
+          scopeWithEventTargets.target !== target
+        ) {
+          continue;
+        }
+
+        if (scopeWithEventTargets.target) {
+          try {
+            scopeWithEventTargets.target.addEventListener =
+              scopeWithEventTargets.eventTargets.addEventListener;
+
+            scopeWithEventTargets.target.removeEventListener =
+              scopeWithEventTargets.eventTargets.removeEventListener;
+
+            delete scopeWithEventTargets.target.__ot_listeners;
+          } catch (e) {
+            console.log(e);
+          }
+        }
+
+        this._scopesWithEventTargets.splice(i, 1);
+
+        if (
+          typeof target !== 'undefined' &&
+          scopeWithEventTargets.target === target
+        ) {
+          break;
+        }
+      }
+    }
+    // console.log('_cleanEventTargetsFromScope2', this._scopesWithEventTargets.length);
```

Comment:
Can this be removed?

Reply:
I would rather keep them as they help me to debug stuff when something is not working - this is for all console.log in this file that are commented out. In worst scenario I can remove them when all is working fine before final merge but would like to keep them for the mentioned reason. Is there any way this could stay as some debugger messages etc ??